generic/zookeepertcl.c

```diff
@@ -1634,25 +1640,29 @@ zootcl_children_subcommand(Tcl_Interp *interp, int objc, Tcl_Obj *CONST objv[],
 
 
 	if (callbackObj == NULL) {
-		// synchronous request
-		// invoke the special async handler for sync requests
-		// so the event loop will still be alive
-		zootcl_syncCallbackContext *zsc = (zootcl_syncCallbackContext *)ckalloc (sizeof (zootcl_syncCallbackContext));
-		zsc->zo = zo;
-		zsc->syncDone = 0;
-		status = zoo_awget_children (zh, path, wfn, watcherCallbackObj, zootcl_sync_strings_completion_callback, zsc);
-		if (zootcl_wait (zo, zsc) == TCL_ERROR) {
-			ckfree (zsc);
-			return TCL_ERROR;
-		}
+		struct String_vector *strings = (struct String_vector *)ckalloc (sizeof (struct String_vector));
+		status = zoo_wget_children(zh, path, wfn, (void *)watcherCallbackObj, strings);	
 
-		if (status != ZOK) {
-			ckfree (zsc);
+		if (status != ZOK && status != ZNONODE) {
+			ckfree (strings);
 			return zootcl_set_tcl_return_code (interp, status);
+		} else if (status == ZNONODE) {
+			// do not consider a non-existent path to be an error in this case
+			status = ZOK;
+			strings->count = 0;
+		}
+		
+		int count = strings ? strings->count : 0;
+		Tcl_Obj **listObjv = (Tcl_Obj **)ckalloc (sizeof(Tcl_Obj *) * count);
```

Comment:
Here and in following lines, you're potentially ckalloc()ing a zero length buffer and passing it to Tcl_NewListObj(). I assume this is known to be safe, but it's a bit surprising, so it seems there should be a different path just setting the result to Tcl_NewObj() if there are zero children?

Reply:
Agreed about `ckalloc`ing a zero length buffer.  I have added an if statement that branches on whether or not `count` is `0`.

Reply:
Roger.